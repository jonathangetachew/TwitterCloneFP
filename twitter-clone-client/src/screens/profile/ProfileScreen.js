import React from 'react';
import { ActivityIndicator, Platform, StyleSheet, View, ScrollView } from 'react-native';
import { Avatar, Text, Icon, Image } from 'react-native-elements';
import { connect } from 'react-redux';
import Colors from '../../constants/Colors';
import RoundButton from '../../components/RoundButton';
import TabBarIcon from '../../components/TabBarIcon';
import TabViewComponent from '../../components/TabViewComponent';
import TweetList from '../../components/TweetList';

const FirstRoute = () => (
    <View style={[styles.scene, { height: 300 }]}>
        <Text>a</Text>
    </View>
);

const SecondRoute = () => (
    <View style={[styles.scene]} />
);

export class ProfileScreen extends React.Component {
    renderActions() {
        const { user } = this.props;
        if (user && user.email) {// Todo: Add condition to check if is current user's profile or not.
            return (
                <View style={styles.actionButtonStyles}>
                    <RoundButton
                        title={"Edit profile"}
                        type="outline"
                    />
                </View>
            );
        }

        return (
            <View style={styles.actionButtonStyles}>
                <RoundButton
                    type="outline"
                    icon={{ name: "mail-outline", color: Colors.primaryColor }}
                />
                <RoundButton
                    type="outline"
                    icon={{ name: "add-alert", color: Colors.primaryColor }}
                />
                <RoundButton
                    title={"Following"}
                />
            </View>
        );
    }
    render() {
        const { background, avatar_url, name, username, description, followers, following } = this.props;
        return (
            <ScrollView>
                <View style={styles.container}>
                    <Image
                        source={{ uri: background || "https://cdn.pixabay.com/photo/2018/08/31/18/21/fantasy-3645269_960_720.jpg" }}
                        style={{ width: '100%', height: 200 }}
                        PlaceholderContent={<ActivityIndicator />}
                    />
                    <Avatar
                        rounded
                        containerStyle={styles.avatarStyle}
                        size={120}
                        source={{
                            uri:
                                avatar_url || 'https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg'
                        }}
                    />
                    {this.renderActions()}
                    <View style={styles.profileInfo}>
                        <Text h3>{name || "Jane Doe"}</Text>
                        <Text style={{ color: '#5c5c5c', fontSize: 20 }}>{username || '@JaneDoe'}</Text>
                        <Text style={{ marginTop: 5, fontSize: 20 }}>{description || 'Sample description'}</Text>
                        <View style={{ flexDirection: 'row', marginTop: 10 }}>
                            <View style={{ flexDirection: 'row', marginEnd: 10 }}>
                                <Text style={styles.numericBold}>{following || 6}</Text>
                                <Text>Following</Text>
                            </View>
                            <View style={{ flexDirection: 'row' }}>
                                <Text style={styles.numericBold}>{followers || 3}</Text>
                                <Text>Followers</Text>
                            </View>
                        </View>
                    </View>
                    <TabViewComponent routes={[
                        { title: 'Tweets', component: TweetList },
                        { title: 'Tweets & replies', component: SecondRoute },
                        { title: 'Media', component: FirstRoute },
                        { title: 'Likes', component: SecondRoute },
                    ]} />
                </View>
            </ScrollView>
        );
    }
}

ProfileScreen.navigationOptions = {
    drawerIcon: () => <Icon size={30} color={Colors.primaryColor} name={"person"} />,
    tabBarIcon: ({ focused }) => <TabBarIcon focused={focused} name={Platform.OS === 'ios' ? 'ios-person' : 'md-person'} />
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'stretch'
    },
    avatarStyle: {
        position: 'absolute',
        left: 5,
        top: 130,
        borderColor: '#fff',
        borderWidth: 2,
        //zIndex: 100,
        backgroundColor: 'red'
    },
    profileInfo: {
        paddingHorizontal: 10
    },
    numericBold: {
        fontWeight: 'bold',
        marginEnd: 5
    },
    actionButtonStyles: {
        alignSelf: 'flex-end',
        marginTop: 10,
        marginRight: 10,
        flexDirection: 'row',
        alignItems: 'center'
    },
    scene: {
        flex: 1,
    }
});

export default connect(({ user }) => ({ user: {...user, email:'current@mail.com'} }))(ProfileScreen);