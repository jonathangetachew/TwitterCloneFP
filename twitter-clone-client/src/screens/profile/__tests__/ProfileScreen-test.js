import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import { ProfileScreen } from '../ProfileScreen';

jest.mock('react-native-elements', () => ({
    Image: 'Image',
    Avatar: 'Avatar',
    Text: 'Text'
}));

jest.mock('../../../components/TabViewComponent', () => 'TabViewComponent');
jest.mock('../../../components/RoundButton', () => 'RoundButton');

describe('ProfileScreen', () => {
    jest.useFakeTimers();

    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it(`renders current user's profile`, () => {
        const tree = renderer.create(<ProfileScreen user={{ email: 'current@mail.com' }} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it(`renders another user's profile`, () => {
        const tree = renderer.create(<ProfileScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
