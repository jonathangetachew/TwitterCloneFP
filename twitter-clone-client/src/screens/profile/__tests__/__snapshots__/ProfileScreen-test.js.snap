// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen renders another user's profile 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flex": 1,
        }
      }
    >
      <Image
        PlaceholderContent={
          <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="small"
          />
        }
        source={
          Object {
            "uri": "https://cdn.pixabay.com/photo/2018/08/31/18/21/fantasy-3645269_960_720.jpg",
          }
        }
        style={
          Object {
            "height": 200,
            "width": "100%",
          }
        }
      />
      <Avatar
        containerStyle={
          Object {
            "backgroundColor": "red",
            "borderColor": "#fff",
            "borderWidth": 2,
            "left": 5,
            "position": "absolute",
            "top": 130,
          }
        }
        rounded={true}
        size={120}
        source={
          Object {
            "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "flexDirection": "row",
            "marginRight": 10,
            "marginTop": 10,
          }
        }
      >
        <RoundButton
          icon={
            Object {
              "color": "#1da1f2",
              "name": "mail-outline",
            }
          }
          type="outline"
        />
        <RoundButton
          icon={
            Object {
              "color": "#1da1f2",
              "name": "add-alert",
            }
          }
          type="outline"
        />
        <RoundButton
          title="Following"
        />
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          h3={true}
        >
          Jane Doe
        </Text>
        <Text
          style={
            Object {
              "color": "#5c5c5c",
              "fontSize": 20,
            }
          }
        >
          @JaneDoe
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginTop": 5,
            }
          }
        >
          Sample description
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginEnd": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                  "marginEnd": 5,
                }
              }
            >
              6
            </Text>
            <Text>
              Following
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                  "marginEnd": 5,
                }
              }
            >
              3
            </Text>
            <Text>
              Followers
            </Text>
          </View>
        </View>
      </View>
      <TabViewComponent
        routes={
          Array [
            Object {
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(TweetList)",
                "type": [Function],
              },
              "title": "Tweets",
            },
            Object {
              "component": [Function],
              "title": "Tweets & replies",
            },
            Object {
              "component": [Function],
              "title": "Media",
            },
            Object {
              "component": [Function],
              "title": "Likes",
            },
          ]
        }
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`ProfileScreen renders current user's profile 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flex": 1,
        }
      }
    >
      <Image
        PlaceholderContent={
          <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="small"
          />
        }
        source={
          Object {
            "uri": "https://cdn.pixabay.com/photo/2018/08/31/18/21/fantasy-3645269_960_720.jpg",
          }
        }
        style={
          Object {
            "height": 200,
            "width": "100%",
          }
        }
      />
      <Avatar
        containerStyle={
          Object {
            "backgroundColor": "red",
            "borderColor": "#fff",
            "borderWidth": 2,
            "left": 5,
            "position": "absolute",
            "top": 130,
          }
        }
        rounded={true}
        size={120}
        source={
          Object {
            "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "flexDirection": "row",
            "marginRight": 10,
            "marginTop": 10,
          }
        }
      >
        <RoundButton
          title="Edit profile"
          type="outline"
        />
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          h3={true}
        >
          Jane Doe
        </Text>
        <Text
          style={
            Object {
              "color": "#5c5c5c",
              "fontSize": 20,
            }
          }
        >
          @JaneDoe
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginTop": 5,
            }
          }
        >
          Sample description
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginEnd": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                  "marginEnd": 5,
                }
              }
            >
              6
            </Text>
            <Text>
              Following
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                  "marginEnd": 5,
                }
              }
            >
              3
            </Text>
            <Text>
              Followers
            </Text>
          </View>
        </View>
      </View>
      <TabViewComponent
        routes={
          Array [
            Object {
              "component": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(TweetList)",
                "type": [Function],
              },
              "title": "Tweets",
            },
            Object {
              "component": [Function],
              "title": "Tweets & replies",
            },
            Object {
              "component": [Function],
              "title": "Media",
            },
            Object {
              "component": [Function],
              "title": "Likes",
            },
          ]
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
